src/home_robot_hw/home_robot_hw/stretch_client/modules/abstract.py

```diff
@@ -0,0 +1,45 @@
+import abc
+
+import rospy
+
+
+def enforce_enabled(func):
+    """Decorator for checking if a control method is executed while module is enabled."""
+
+    def wrapper(self, *args, **kwargs):
+        if self.is_enabled:
+            return func(self, *args, **kwargs)
+        else:
+            err_str = f"{type(self).__name__}.{func.__name__} is only available in when the corresponding control mode is active."
+            rospy.logerr(err_str)
+            raise TypeError(err_str)
+
+    return wrapper
+
+
+class AbstractControlModule(abc.ABC):
```

Comment:
Should this be outside hello stretch hw?

Reply:
i think it could be in interfaces since technically doesnt need ROS

Reply:
It doesn't directly depend on ROS, but it is also not used by anything outside of the hardware stack (nor will it be anytime in the near future), so I'd like to keep it in home_robot_hw. Happy to discuss.